app/src/main/java/com/powerhungers/fomezero/feature/registration/di/RegistrationModule.kt

```diff
@@ -0,0 +1,13 @@
+package com.powerhungers.fomezero.feature.registration.di
+
+import com.powerhungers.fomezero.feature.registration.domain.usecase.RegistrationUseCase
+import com.powerhungers.fomezero.feature.registration.presentation.RegistrationViewModel
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+
+val registrationModule = module {
+
+    viewModel { RegistrationViewModel(
+        registrationUseCase = RegistrationUseCase(repository = get())
+    ) }
```

Comment:
```suggestion
 viewModel {
     RegistrationViewModel(
        registrationUseCase = RegistrationUseCase(
             repository = RegistrationRepositoryImpl(
                  remoteDataSource = RegistrationRemoteDataSource(
                      service = get<Retrofit>().create(RegistrationService::class.java)
                  )
             )
         )
    )
}
```
a identação dessa injeção de dependencia esta incorreta, se usar o ctrl+alt+l ela ja se corrige.
Outro ponto é caso o repository data source e service sejam utilizados somente por uma viewmodel é possivel fazer a injeção toda em somente um factory de viewModel isso otimiza o projeto pois a arvore de injeçoes fica menor e nao faz injeçoes alem do necessario

Reply:
se vc nao tiver criado a injeçao do repository separado e usar esse get() a injeção nao ira encontrar a referencia do repository e o app vai crashar, e no logcat vai acusar que o inject da viewModel esta incorreto, mas na realidade é o get que nao ta encontrando a referencia